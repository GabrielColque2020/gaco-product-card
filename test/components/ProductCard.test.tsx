import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { ProductCard } from "../../src/";
import { product1 } from "../data/products";


describe( 'ProductCard', () => {
	test( 'debe de mostrar el componente correctamente', () => {
		const rerender = render(
			<ProductCard product={ product1 }>
				{
					() => (
						<h1>Product Card</h1>
					)
				}
			</ProductCard>
		);
		expect( rerender.baseElement ).toMatchSnapshot();
	} );

	test( 'debe de incrementar el contador', () => {
		const rerender = render(
			<ProductCard product={ product1 }>
				{
					( { count, increaseBy } ) => (
						<>
							<h1>Product Card</h1>
							<span>{ count }</span>
							<button onClick={ () => increaseBy( 1 ) }>Increment</button>
						</>
					)
				}
			</ProductCard>
		);


		expect( rerender.baseElement ).toMatchSnapshot();

		const button = rerender.getByText('Increment');
		fireEvent.click(button);

		expect( rerender.baseElement ).toMatchSnapshot();
	} );
} );