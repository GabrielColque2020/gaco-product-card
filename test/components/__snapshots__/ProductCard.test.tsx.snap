// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard debe de incrementar el contador 1`] = `
<body>
  <div>
    <div
      class="productCard undefined"
    >
      <h1>
        Product Card
      </h1>
      <span>
        0
      </span>
      <button>
        Increment
      </button>
    </div>
  </div>
</body>
`;

exports[`ProductCard debe de incrementar el contador 2`] = `
<body>
  <div>
    <div
      class="productCard undefined"
    >
      <h1>
        Product Card
      </h1>
      <span>
        1
      </span>
      <button>
        Increment
      </button>
    </div>
  </div>
</body>
`;

exports[`ProductCard debe de mostrar el componente correctamente 1`] = `
<body>
  <div>
    <div
      class="productCard undefined"
    >
      <h1>
        Product Card
      </h1>
    </div>
  </div>
</body>
`;

exports[`ProductImage debe de mostrar el componente con la imagen del producto 1`] = `
<body>
  <div>
    <div
      class="productCard undefined"
    >
      <img
        alt="Product Image"
        class="productImg undefined"
        src="./coffee-mug.png"
      />
    </div>
  </div>
</body>
`;

exports[`ProductImage debe de mostrar el componente correctamente con la imagen personalizada 1`] = `
<body>
  <div>
    <img
      alt="Product Image"
      class="productImg undefined"
      src="https://hola.jpg"
    />
  </div>
</body>
`;
